runtime/extern.c

```diff
@@ -939,7 +1057,7 @@ static intnat extern_value(struct caml_extern_state* s, value v, value flags,
 
 void caml_output_val(struct channel *chan, value v, value flags)
 {
-  char header[32];
+  char header[MAX_INTEXT_HEADER_SIZE];
   int header_len;
   struct output_block * blk, * nextblk;
   struct caml_extern_state* s = get_extern_state ();
```

Comment:
While reviewing the code here I found an unrelated bug that would be good to be fixed separately, opened an issue here: https://github.com/ocaml/ocaml/issues/12037